.github/workflows/base.yml

```
@@ -0,0 +1,45 @@
+name: base
+
+on:
+  schedule:
+    - cron: '0 15 * * *'
+  pull_request:
+    paths:
+      - './**'
+      - '.github/workflows/base.yml'
+
+jobs:
+  examples:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: setup-python${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install (Python)
+      run: |
+        python -m pip install --upgrade pip
+        pip install --progress-bar off -U setuptools
+        pip install git+https://github.com/optuna/optuna.git
+        python -c 'import optuna'
+
+        pip install -r requirements.txt
+        # [todo crissman] resolve cudf issues for rapids
```

Not a strong opinion but this is the format we often use that's also recognized by flake8.
```suggestion
        # TODO(crissman): resolve cudf issues for rapids
```